Subject: timer:fix build warning in timer.c
From: Shi Weihua <shiwh@cn.fujitsu.com>
Date: Thu, 14 Feb 2008 18:02:14 +0800

    [ The following text is in the "UTF-8" character set. ]
    [ Your display is set for the "iso-8859-1" character set.  ]
    [ Some characters may be displayed incorrectly. ]

Fix the following compile warning without CONFIG_PREEMPT_RT:
kernel/timer.c:937: warning: ^^count_active_rt_tasks^^ defined but not used

Signed-off-by: Shi Weihua <shiwh@cn.fujitsu.com>

Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
---
 kernel/timer.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

Index: linux-2.6-tip/kernel/timer.c
===================================================================
--- linux-2.6-tip.orig/kernel/timer.c
+++ linux-2.6-tip/kernel/timer.c
@@ -1199,20 +1199,18 @@ static unsigned long count_active_tasks(
 #endif
 }
 
+#ifdef CONFIG_PREEMPT_RT
 /*
  * Nr of active tasks - counted in fixed-point numbers
  */
 static unsigned long count_active_rt_tasks(void)
 {
-#ifdef CONFIG_PREEMPT_RT
 	extern unsigned long rt_nr_running(void);
 	extern unsigned long rt_nr_uninterruptible(void);
 
 	return (rt_nr_running() + rt_nr_uninterruptible()) * FIXED_1;
-#else
-	return 0;
-#endif
 }
+#endif
 
 /*
  * Hmm.. Changed this, as the GNU make sources (load.c) seems to
@@ -1241,7 +1239,9 @@ static inline void calc_load(unsigned lo
 	count -= ticks;
 	if (unlikely(count < 0)) {
 		active_tasks = count_active_tasks();
+#ifdef CONFIG_PREEMPT_RT
 		active_rt_tasks = count_active_rt_tasks();
+#endif
 		do {
 			CALC_LOAD(avenrun[0], EXP_1, active_tasks);
 			CALC_LOAD(avenrun[1], EXP_5, active_tasks);
